src/EnergyPlus/AirflowNetworkSolver.cc

```diff
@@ -3666,6 +3686,156 @@ namespace AirflowNetworkSolver {
         }
     }
 
+    void GenericDuct(Real64 const Length,   // Duct length
+                     Real64 const Diameter, // Duct diameter
+                     int const LFLAG,       // Initialization flag.If = 1, use laminar relationship
+                     Real64 const PDROP,    // Total pressure drop across a component (P1 - P2) [Pa]
+                     int const n,           // Node 1 number
+                     int const M,           // Node 2 number
+                     Array1A<Real64> F,     // Airflow through the component [kg/s]
+                     Array1A<Real64> DF,    // Partial derivative:  DF/DP
+                     int &NF                // Number of flows, either 1 or 2
+    )
+    {
+
+        // This subroutine solve air flow as a duct if fan has zero flow rate
+
+        // Argument array dimensioning
+        F.dim(2);
+        DF.dim(2);
+
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        Real64 const C(0.868589);
+        Real64 const EPS(0.001);
+        Real64 const Rough(0.0001);
+        Real64 const InitLamCoef(128.0);
+        Real64 const LamDynCoef(64.0);
+        Real64 const LamFriCoef(0.0001);
+        Real64 const TurDynCoef(0.0001);
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 A0;
+        Real64 A1;
+        Real64 A2;
+        Real64 B;
+        Real64 D;
+        Real64 S2;
+        Real64 CDM;
+        Real64 FL;
+        Real64 FT;
+        Real64 FTT;
+        Real64 RE;
+        Real64 ed;
+        Real64 ld;
+        Real64 g;
+        Real64 AA1;
+        Real64 area;
```

Comment:
These should all be defined right where they are used, and defined as `const` whenever possible.

Reply:
@Myoldmopar Done.